perspective_reddit_bot/check_mod_actions.py

```diff
@@ -23,50 +23,70 @@
 import json
 import praw
 import time
+import sys
 
 from creds import creds
 from log_subreddit_comments import append_record, now_timestamp
 
 
 APPROVED_COL = 'approved'
 REMOVED_COL = 'removed'
+DELETED_COL = 'deleted'
 
 
 def write_moderator_actions(reddit,
-                            line,
+                            record,
                             id_key,
                             timestamp_key,
                             output_path,
-                            hours_to_wait):
-  record = json.loads(line)
+                            hours_to_wait,
+                            has_mod_creds):
   bot_scored_time = datetime.strptime(record[timestamp_key], '%Y%m%d_%H%M%S')
   time_to_check = bot_scored_time + timedelta(hours=hours_to_wait)
   wait_until(time_to_check)
-  approved_removed = check_approved_removed(reddit, record[id_key])
-  if approved_removed:
-    record[APPROVED_COL], record[REMOVED_COL] = approved_removed
-  else:
-    record[APPROVED_COL] = None
-    record[REMOVED_COL] = None
+
   record['action_checked_utc'] = now_timestamp()
+  status_fields = check_comment_status(reddit, record[id_key], has_mod_creds)
+  if status_fields is not None:
+    record.update(status_fields)
+
   append_record(output_path, record)
 
 
-def check_approved_removed(reddit, comment_id):
+def check_comment_status(reddit, comment_id, has_mod_creds):
```

Comment:
It's a little confusing to have 2 functions, `check_comment_status` and `get_comment_status` that in a non-error case return the same thing with just slightly different args.  Maybe you can consolidate these into 1 function?  Or rename the function to be more explicit about the difference?

Reply:
It's split this way so that it's easier to test the logic.

I renamed `check_comment_status` to `fetch_reddit_comment_status`.